### ISSUE TITLE
MeshAgent Desktop is not connecting for Linux RHEL version7x and 8x machines.

### ISSUE THREAD

**Issue by Abarna512** [Ylianst/MeshCentral#5262]
MeshAgent Desktop is not connecting properly for Linux RHEL version7x and 8x machines.

We have tried this below workaround and remote desktop was working fine for few days.

1. Once meshagent is installed in Linux, check the remote desktop access in meshportal.
2. If it is not connecting or showning blank then follow the below steps.
3. Go to the /etc/gdm/ path if present or go to /etc/gdm3/ path
4. Open custom.conf file
5. Under [daemon] tag, add the below lines if not present.(if the lines are commented (#) then uncomment the lines)
    WaylandEnable=false
    DefaultSession=gnome-xorg.desktop
6. Now execute the below commands 
    service gdm restart
    service gdm3 restart

Then after few days it was not connecting. We have executed **service gdm restart** in terminal and it started working again.
Is there any solution to fix this issue without executing **service gdm restart** command again and again?
Meshcentral version 1.0.0

![dccb](https://github.com/Ylianst/MeshCentral/assets/140589141/b689dada-f48f-408f-ba1f-9904ce8df3fe)

**Comment by si458** [comment]
sadly meshcentral doesnt work correctly with Wayland so the `WaylandEnable=false` is required!
as for why it stops working after so long and you need to restart gdm, that seems weird?
have you tried without the `DefaultSession=gnome-xorg.desktop` line being set?

**Comment by Abarna512** [comment]
Yes, we have tried without setting **DefaultSession=gnome-xorg.desktop**, And it was not working. 

In few machines, desktop was connecting after doing the DefaultSession=gnome-xorg.desktop configuration.  
But again, it stopped working. So that we executed the **gdm restart** command.

Can we make it work without giving gdm restart command?

**Comment by silversword411** [comment]
If this is an ongoing issue, and you're willing to pay to have this fixed I'd recommend:

1. Set up a specific description and thing you want fixed
2. Put together a reasonable development budget and see if you can find a developer willing to work on and implement it.
3. Contact @Ylianst and see if he's the developer that will do the work you can provide a specific payout requirement for what you want (since he's the most competent person available)
4. If he can't help, then open up the opportunity to others, use a system like GoFundMe or something that will provide public accountability for a development service and a payment for that service.
